### PR TITLE
changing username by admin

### PR DESCRIPTION
**Pull request by soonahn**
Addresses https://github.com/RocketChat/Rocket.Chat/issues/698

### CODE REVIEW

**Comment by geekgonecrazy** [comment]
This also looks solid.  Going to test both this and #696.  The underlying method hasn't been tested fully.

File: packages/rocketchat-lib/server/methods/updateUser.coffee [hunk 1]
```diff
@@ -16,5 +16,9 @@ Meteor.methods
 		unless userData.name
 			throw new Meteor.Error 'name-is-required', 'Name field is required'
 
+		unless userData.username
+			throw new Meteor.Error 'user-name-is-required', 'Username field is required'
+
 		Meteor.users.update { _id: userData._id }, { $set: { name: userData.name } }
+		Meteor.users.update { _id: userData._id }, { $set: { username: userData.username } }
```
Comment (engelgabriel) [inline packages/rocketchat-lib/server/methods/updateUser.coffee]: It should be just:
```
Meteor.users.update { _id: userData._id }, { $set: { name: userData.name, username: userData.username } }
```
in a single update.

**Comment by engelgabriel** [comment]
Thanks for the PR. Can you please just fix the line above?

**Comment by engelgabriel** [comment]
Guys, update the username is not that simple. You need to use the method. There are a lot of checking and references that need o be updated together, not just the users collection.

**Comment by engelgabriel** [comment]
See /packages/rocketchat-lib/server/methods/setUsername.coffee

**Comment by soonahn** [comment]
Updated so the update method is only called once.  New to meteor, didn't realise that was possible.  Thanks!

**Comment by engelgabriel** [comment]
I think this should be done, but using the right method call

**Comment by soonahn** [comment]
@engelgabriel fixed to use the setUsername method.

**Pull request merged**